src/storage/table/standard_column_data.cpp

```diff
@@ -227,8 +227,19 @@ unique_ptr<ColumnCheckpointState> StandardColumnData::Checkpoint(RowGroup &row_g
 	// to prevent reading the validity data immediately after it is checkpointed we first checkpoint the main column
 	// this is necessary for concurrent checkpointing as due to the partial block manager checkpointed data might be
 	// flushed to disk by a different thread than the one that wrote it, causing a data race
-	auto base_state = ColumnData::Checkpoint(row_group, checkpoint_info);
-	auto validity_state = validity.Checkpoint(row_group, checkpoint_info);
+
+	// Always grab the base lock first
+	auto base_lock = GetSegmentLock();
```

Comment:
A better solution here would be to unify the checkpoint so that we no longer do two separate checkpoints (column data followed by validity). This improves on the current design, since we always scan the validity twice currently, which we then would not need to do - in addition to solving the issue outlined here.